include/geometric_shapes/check_isometry.h

```diff
@@ -0,0 +1,118 @@
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, Martin Pecka
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Martin Pecka, Robert Haschke */
+
+#ifndef GEOMETRIC_SHAPES_CHECK_ISOMETRY_H
+#define GEOMETRIC_SHAPES_CHECK_ISOMETRY_H
+
+#include <algorithm>
+#include <assert.h>
+#include <iostream>
+#include <math.h>
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+/** \brief This file provides functions and macros that can be used to verify that an Eigen::Isometry3d is really an
+ * isometry. Eigen itself doesn't do the checks because they're expensive. If the isometry object is constructed in
+ * a wrong way (e.g. from an AngleAxisd with non-unit axis), it can represent a non-isometry. But some methods in the
+ * Isometry3d class perform isometry-specific operations which return wrong results when called on a non-isometry. E.g.
+ * for isometries, .linear() and .rotation() should be the same, but for non-isometries, the result of .linear()
+ * contains also the scaling factor, whether .rotation() is only the rotation part.
+ *
+ * These checks are primarily meant to be performed only in debug mode (via the ASSERT_ISOMETRY macro), but you can call
+ * checkIsometry() even in release mode. This check should be mainly performed on transforms input by the user.
+ */
+namespace bodies
```

Comment:
I don't see why this generic function should live in `bodies::`.
Any complains about moving this out of the namespace?

Reply:
I think having it in no namespace would be dangerous... But a different namespace might work... I though about misusing `Eigen`, but I'm not sure it's a good idea.

Reply:
Given the generic nature of this function, I don't see a problem with the global namespace. Hijacking `Eigen` is wrong. Going for a `utils` namespace is not much better than the global namespace, `geometric_shapes::checkIsometry` seems rather long (but still better than the current version).
Either way, the main interface of this code, `ASSERT_ISOMETRY`, of course does not use a namespace..

Reply:
> Either way, the main interface of this code, `ASSERT_ISOMETRY`, of course does not use a namespace..

That's a fair argument, we're polluting the global namespace anyways...

Reply:
I removed the namespace.